### PR TITLE
Support for Type hint

### PR DESCRIPTION
**Pull request by letientai299**
- Implement the feature with tests. 
- Update doc.
- Correct some small issues in doc (change from `template_dir_template_dir` to `pydocstring_templates_dir`)

### CODE REVIEW

**Comment by heavenshell** [comment]
@letientai299 Thanks I'll look into it. Wait for a while 😅

File: test/type-hint.vader [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        pass
+
+Execute:
+    Pydocstring
+
+Expect python:
+    def foo(n: int, m: str) -> int:
+        """foo
+
+        :param n (int):
```
Comment (heavenshell) [inline test/type-hint.vader]: Currently pydocstring.vim's default style is Sphinx style.
Can we do like following?
```python
"""foo

:param n:
:type n: int
:param m:
:type m: str
:rtype: int
```

**Comment by letientai299** [comment]
Hi @heavenshell, how do you think about my update 😄 ?

**Comment by heavenshell** [comment]
Sorry for late response.
I'll look into this weekend.

**Comment by heavenshell** [comment]
@letientai299 
Merged. Thank you for a great PR!!
Awsome!

**Pull request merged**